src/viam/sdk/rpc/server.cpp

```diff
@@ -1,39 +1,69 @@
 #include <viam/sdk/rpc/server.hpp>
 
-#include <grpcpp/ext/proto_server_reflection_plugin.h>
+#include <sstream>
+
+#include <boost/log/trivial.hpp>
 #include <grpcpp/security/server_credentials.h>
 
+#include <viam/sdk/registry/registry.hpp>
+
 namespace viam {
 namespace sdk {
 
-Server::Server() : builder_(std::make_unique<grpc::ServerBuilder>()) {}
+Server::Server() : builder_(std::make_unique<grpc::ServerBuilder>()) {
+    Registry::initialize();
+    for (const auto& rr : Registry::registered_resources()) {
+        auto new_manager = std::make_shared<ResourceManager>();
+        auto server = rr.second->create_resource_server(new_manager, *this);
+        managed_servers_.emplace(server->api(), std::move(server));
```

Comment:
Register all known resources when creating the server, to ensure proper gRPC reflection.

I don't like having this logic (Registry initializing, creating of managed resource servers) happening inside the Server constructor. But, it does have the nice property of creating only a single point where all this is handled, and not requiring users to call initialize themselves (which could be a bit of a gotcha).

I tried moving this logic into the `Server::start()` call where I think it makes a bit more sense, but this created a problem where it was impossible to add a managed resource until the server had already started. This made it impossible, e.g., for the `RobotService` to make sure that all its managed resources were taken care of at construction. Given that this is relatively hidden from users (they should never have to deal with registration at this level, or deal with the look of this implementation), I thought the usefulness properties and lack of user-facing issues outweighed the slight code-smell that comes from it. Happy to revisit if others disagree.

Reply:
Yeah I'm really feeling the lack of clarity around what the root of initialization and ownership is in the SDK. I don't think we are going to get there in this review.

Reply:
😞 yeah that is fair. I suspect a second PR that focuses broadly on this specific question will be important in the near future.